lib/Mojolicious/Plugin/CGI.pm

```diff
@@ -278,15 +282,15 @@ sub _stdout_cb {
     $buf .= $chunk;
     $buf =~ s/^(.*?\x0a\x0d?\x0a\x0d?)//s or return;    # false until all headers has been read from the CGI script
     $headers = $1;
-
-    if ($headers =~ /^HTTP/) {
+    if ($headers =~ /^HTTP(?: (\d\d\d))?/) {
+      $c->res->code($1) if $1;
```

Comment:
You mean, why is the status code optional? You are right – it should not be. RFC 2616 section 6.1 is quite clear about it: `Status-Line = HTTP-Version SP Status-Code SP Reason-Phrase CRLF`. I was simply afraid of breaking something because the old code made it optional. So I'll be using the following:
```
if ($headers =~ /^HTTP (\d\d\d)/) {
```